app/models/user.rb

```diff
@@ -1,6 +1,8 @@
 class User < ApplicationRecord
   belongs_to :provider
 
+  has_many :trainees, through: :provider
```

Comment:
I'd be keen to avoid defining this association unless we need it as it doesn't feel right for the domain. I get that you've used it in the controller actions so we can leave it for now as long as we remember to remove it when we switch over to using the policies. Just conscious that TT-API is a nest of unwarranted or unused associations and don't want to recreate that.